kolibri/plugins/user/assets/src/routes.js

```diff
@@ -1,91 +1,120 @@
 import store from 'kolibri.coreVue.vuex.store';
 import router from 'kolibri.coreVue.router';
+import Lockr from 'lockr';
+import { SIGNED_OUT_DUE_TO_INACTIVITY } from 'kolibri.coreVue.vuex.constants';
 import { showSignInPage } from './modules/signIn/handlers';
 import { showSignUpPage } from './modules/signUp/handlers';
 import { showProfilePage } from './modules/profile/handlers';
-import { PageNames } from './constants';
+import { ComponentMap, PageNames } from './constants';
+
+import AuthSelect from './views/AuthSelect';
+import FacilitySelect from './views/FacilitySelect';
+import ProfilePage from './views/ProfilePage';
+import SignInPage from './views/SignInPage';
+import SignUpPage from './views/SignUpPage';
+
+router.beforeEach((to, from, next) => {
+  const profileRoutes = [ComponentMap.PROFILE, ComponentMap.PROFILE_EDIT];
+  // If logged in and not already going to a Profile page, redirect
+  if (store.getters.isUserLoggedIn && !profileRoutes.includes(to.name)) {
+    next(router.getRoute(ComponentMap.PROFILE));
+  } else {
+    next();
+  }
+});
 
 export default [
   {
-    name: PageNames.ROOT,
     path: '/',
-    handler: () => {
+    name: 'root',
+    beforeEnter(to, from, next) {
+      // Always redirect to Profile if logged in.
       if (store.getters.isUserLoggedIn) {
-        router.replace({ name: PageNames.PROFILE });
+        next(router.getRoute(ComponentMap.PROFILE));
       } else {
-        // AUTH_SELECT for multiple facilities device
+        // If Multiple Facilities but we've not stored a facilityId in localstorage
+        // then we go to the AuthSelect route
         if (store.getters.facilities.length > 1 && !store.state.facilityId) {
-          router.replace({ name: PageNames.AUTH_SELECT });
+          next(router.getRoute(ComponentMap.AUTH_SELECT));
         } else {
-          router.replace({ name: PageNames.SIGN_IN });
+          next(router.getRoute(ComponentMap.SIGN_IN));
```

Comment:
Is the `ComponentMap` necessary? I think you can reference these directly like:

```suggestion
          next(router.getRoute(SignInPage.name));
```

Reply:
In routes.js it would make sense since they're all imported in that file anyway, but the ComponentMap comes in handy within components as well.

I guess the difference is that `import { ComponentMap } from './constants'` in a bunch of files is one import per file - but if I needed to get the route for two or more components from one place then I'd have to import them all, even if they're not used in the template but only as places to be routed to.